.gitignore

```diff
@@ -5,8 +5,14 @@
 build
 target
 bin
+
+# ignore all Idea files except for common codestyle settings
 *.iml
-.idea/
+.idea/*
+!.idea/checkstyle-idea.xml
+!.idea/codeStyles/
```

Comment:
is it style correct?

Reply:
What do you mean? this is not a .java file.

Reply:
as long as you have this directory with files it will not be ignored by git. AFICS we do not add anything from this directory `.idea/codeStyles/` we have only 2 files to not ignore `.idea/checkstyle-idea.xml ` and `.idea/google-java-format.xml`

Reply:
Ah, you meant this line is extra, good catch! leftover from the other approach.

Reply:
Fixed!